src/modules/rewards/components/RewardsTabsContainer.tsx

```diff
@@ -32,6 +35,10 @@ const RewardsTabsContainer: FC<RewardsTabsContainerProps> = ({
     if (activeTab === 'activity' && userPushSDKInstance && userPushSDKInstance.readmode()) {
       setShowConnectModal(true);
     }
+
+    if (activeTab !== 'activity' && userDetails) {
+      setShowConnectModal(false);
+    }
   }, [activeTab, userPushSDKInstance]);
```

Comment:
Dude activityTab, userPushSDKInstance, setShowConnectModal and userDetails all are present in Rewards.tsx. Why are you passing down all the props here and applying this useEffect here?

This whole useEffect needs to go to a custom hook called useRewardsConnectModal and will be called in Rewards.tsx.

The whole idea of our previous discussion was to keep things at the correct places. This stuff needs to happen on topmost level in Rewards.tsx